src/GNDStk/Node/test/all.test.cpp

```diff
@@ -1,26 +1,96 @@
 
 #include "catch.hpp"
 #include "GNDStk.hpp"
+#include <cstring>
 
 using namespace njoy::GNDStk;
 using namespace mixed::child;
+using namespace mixed::meta;
 
 SCENARIO("Testing GNDStk Node all()") {
 
    GIVEN("A tree read from a particular example GNDS file") {
       // tree
       const Tree<> tree("n-008_O_016.xml");
 
-      // node: chemicalElements
-      const auto &elems = tree(reactionSuite,PoPs,chemicalElements);
+      // chemicalElements / chemicalElement
+      WHEN("We extract all <chemicalElement> nodes within <chemicalElements>") {
+         const auto &CEs = tree(reactionSuite,PoPs,chemicalElements);
+         {
+            // without "found"
+            auto ce = CEs.all("chemicalElement");
+            THEN("We find eight of them") {
+               CHECK(ce.size() == 8);
+            }
+         } {
+            // with "found"
+            bool found = false;
+            auto ce = CEs.all("chemicalElement",found);
+            THEN("We find eight of them") {
+               CHECK(found);
+               CHECK(ce.size() == 8);
+            }
+         }
+      }
+
+      // reactions / reaction
+      WHEN("We extract selected <reactions> nodes within <reaction>") {
+         const auto &Rs = tree(reactionSuite,reactions);
+
+         // filter that requires: node has metadata label="n + *"
+         auto nplus = [](const node &n)
+            { return 0 == strncmp(n(label).c_str(), "n + ", 4); };
```

Comment:
I'm not sure what `label` is here. It looks like a variable, but I can't seem to figure out what it is.

Reply:
It's in GNDStk. Basically,

```
const meta_t<std::string> label("label");
```

So, `label` means: look in the node (in this case called `n`), for the metadatum key/value pair with a key of `"label"`, and return the key's associated value as a `std::string`.